docs/testplan/IPv4-Port-Based-DHCP-Server-test-plan.md

```diff
@@ -0,0 +1,258 @@
+            - [Common setup](#common-setup)
+            - [Common teardown](#common-teardown)
+            - [test_dhcp_server_stress](#test_dhcp_server_stress)
+        - [Test Module #4 test_dhcp_server_smart_switch.py](#test-module-4-test_dhcp_server_smart_switchpy)
+            - [Common setup](#common-setup)
+            - [Common teardown](#common-teardown)
+            - [test_dhcp_server_smart_switch](#test_dhcp_server_smart_switch)
+
+<!-- /TOC -->
+## Related Documents
+
+| **Document Name** | **Link** |
+|-------------------|----------|
+| IPv4 Port Based DHCP_SERVER in SONiC | [port_based_dhcp_server_high_level_design.md](https://github.com/sonic-net/SONiC/blob/master/doc/dhcp_server/port_based_dhcp_server_high_level_design.md)|
+|Smart Switch IP address assignment| [smart-switch-ip-address-assignment.md](https://github.com/sonic-net/SONiC/blob/master/doc/smart-switch/ip-address-assigment/smart-switch-ip-address-assignment.md)|
+
+## Overview
+
+A DHCP Server is a server on network that can automatically provide and assign IP addresses, default gateways and other network parameters to client devices. Port based DHCP server is to assign IPs based on interface index.
+
+## Scope
+
+### Test Scenario
+
+The tests will include:
+
+1. Configuration test
+   1. Add related configuration into CONFIG_DB and then verify configuration and process running status.
+   2. Update related tables in CONFIG_DB to see whether configuration for DHCP Server change too.
+2. Functionality test
+   1. Check whether dhcrelay in dhcp_relay can foward DHCP packets between client and dhcp_server container.
+   2. Check whether dhcp_server container can reply DHCP reply packets as expected.
+   3. Verify in multi-vlan scenario.
+
+### Supported Topology
+
+Base dhcp_server functionality tests (test module [#1](#test-module-1-test_dhcp_serverpy) [#2](#test-module-2-test_dhcp_server_multi_vlanpy) [#3](#test-module-3-test_dhcp_server_stresspy)) are supported on mx topology, smart switch related test (test module [#4](#test-module-4-test_dhcp_server_smart_switchpy)) is supported on t1-smartswitch topology (A new topology on real smart switch testbed).
+
+## Test Case
+
+### Test Module #1 test_dhcp_server.py
+
+#### Port Based Common setup
+
+* Check whether dhcrelay process running as expected (Original dhcp_relay functionality).
+* Enable dhcp_server feature, and then use CLI to add DHCP Server configuration.
+
+#### Port Based Common teardown
+
+* Disable dhcp_server feature, and then check whether dhcrelay process running as expected (Original dhcp_relay functionality).
+* Config reload, remove dhcp_server container.
+
+#### test_dhcp_server_port_based_assignment_single_ip
+
+* **Test objective**
+
+   To test port based single ip assign.
+
+* **Test detail**
+
+  * Send DHCP discover packets from ptf, check whether configured port receive DHCP offer packet and no-configured ports don't receive. Need to check netmask / gateway / lease_time / yiaddr.
```

Comment:
Can we add test setup:

case 1:
Switch does not has client MAC in FDB

case 2:
Switch has client MAC in FDB

case 3:
Switch has client MAC in FDB, but it was learnt from another interface (mac move)

Reply:
Added in `test_dhcp_server_port_based_assignment_single_ip`